src/hapi/pipelines/database/national_risk.py

```diff
@@ -0,0 +1,95 @@
+"""Functions specific to the national risk theme."""
+
+from logging import getLogger
+from typing import Dict
+
+from hapi_schema.db_national_risk import DBNationalRisk
+from hdx.utilities.dictandlist import dict_of_dicts_add
+from sqlalchemy.orm import Session
+
+from . import locations
+from .base_uploader import BaseUploader
+from .metadata import Metadata
+
+logger = getLogger(__name__)
+
+
+class NationalRisk(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        metadata: Metadata,
+        locations: locations.Locations,
+        results: Dict,
+    ):
+        super().__init__(session)
+        self._metadata = metadata
+        self._locations = locations
+        self._results = results
+
+    def populate(self):
+        logger.info("Populating national risk table")
+        for dataset in self._results.values():
+            datasetinfo = self._metadata.runner.scrapers[
+                "national_risk_national"
+            ].datasetinfo
+            reference_period_start = datasetinfo["source_date"][
+                "default_date"
+            ]["start"]
+            reference_period_end = datasetinfo["source_date"]["default_date"][
+                "end"
+            ]
```

Comment:
@mcarans The way I'm parsing the dates from datasetinfo is convoluted but I don't think dates from a config make it into the hapi results - is this something we should consider incorporating?  I can imagine wanting to specify the dates for other datasets.

Reply:
Yes, it will need to be added to HDX Python Scraper, but I'll approve this and we can make the change here later